scripts/build_msan_libcxx.sh

```
@@ -0,0 +1,86 @@
+    git clone --depth=1 https://github.com/llvm/llvm-project.git "$SRC_DIR"
+else
+    echo "llvm-project source already present at $SRC_DIR"
+fi
+
+# Patch libunwind's CMakeLists.txt so LIBUNWIND_ADDITIONAL_COMPILE_FLAGS is
+# applied PRIVATE rather than PUBLIC. The default PUBLIC propagation causes
+# our -fno-sanitize=memory (added to keep libunwind from MSAN-recursing in
+# the reporter) to leak into libcxx via target dependencies, silently
+# disabling MSAN for libc++.so. Idempotent: only patches if not already done.
+LIBUNWIND_CMAKE="$SRC_DIR/libunwind/src/CMakeLists.txt"
+if grep -q 'PUBLIC "${LIBUNWIND_ADDITIONAL_COMPILE_FLAGS}"' "$LIBUNWIND_CMAKE"; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)